pkg/ingester/ingester_v2.go

```diff
@@ -1504,12 +1507,20 @@ func (i *Ingester) createTSDB(userID string) (*userTSDB, error) {
 
 	blockRanges := i.cfg.BlocksStorageConfig.TSDB.BlockRanges.ToMilliseconds()
 
+	// Min of nonzero options
+	maxExemplars := 0
+	if i.limiter != nil {
+		maxExemplars = i.limiter.maxExemplarsPerUser(userID)
+	}
+	maxExemplars = minNonZero(maxExemplars, i.cfg.BlocksStorageConfig.TSDB.MaxExemplars)
```

Comment:
might be useful to have a comment here `maxExemplars is the minimum non-zero value between the per user exemplars limit, the global per user exemplars limit (if shuffle sharding is enabled), and the global exemplar limit for the ingester`

Should we set `maxExemplars` to 0 if `i.cfg.BlocksStorageConfig.TSDB.MaxExemplars == 0` ? Do we want to treat that value like a feature flag?

Reply:
Yes would appreciate feedback on what the interaction between the 3 limits (tsdb, local, global) should look like.   I believe this case is different because exemplars must default to 0 instead of unbounded like other limits.  

Regarding whether TSDB.MaxExemplars should behave like a feature flag:   my thinking was that in a multi-tenant scenario it seemed clearer operationally to enable exemplars using just 1 value anywhere instead of requiring 2.   If a non-zero `TSDB.MaxExemplars` value is also required, it must be larger than the per-tenant limits, i.e. it represents the maximum exemplars that could ever be configured for a single tenant/ingester.   I'm not sure how useful that kind of safeguard is, but if the consensus is that it would be useful, no concerns.

Additionally, I was thinking that for single-tenant or test cortex installations, being able to enable exemplars through only `TSDB.MaxExemplars` was useful (conceptually and operationally), without requiring the usage of a runtime overrides file and the `fake` tenant in what might otherwise be a very straightforward installation.